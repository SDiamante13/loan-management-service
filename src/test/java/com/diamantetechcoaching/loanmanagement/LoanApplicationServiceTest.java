package com.diamantetechcoaching.loanmanagement;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
class LoanApplicationServiceTest {

    @Autowired
    LoanApplicationService loanApplicationService;

    @Test
    void processLoanApplication() {
        LoanApplicationRequest request = new LoanApplicationRequest(
                "first",
                "last",
                10000,
                1000,
                30000,
                "123");

        LoanApplicationResponse response = loanApplicationService.processLoanApplication(request, 750, entity1 -> {});

        assertThat(response.getStatus()).isEqualTo("Approved");
    }
}