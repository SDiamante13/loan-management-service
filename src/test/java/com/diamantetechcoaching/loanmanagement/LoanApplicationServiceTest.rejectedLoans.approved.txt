Rejected Loans[0] = LoanApplicationResponse{status='Rejected', creditScore=700, monthlyIncome=5000.0, monthlyDebt=1500.0, requestedAmount=18000.0, dti=30.0}
Rejected Loans[1] = LoanApplicationResponse{status='Rejected', creditScore=750, monthlyIncome=5000.0, monthlyDebt=10000.0, requestedAmount=18000.0, dti=200.0}
Rejected Loans[2] = LoanApplicationResponse{status='Rejected', creditScore=750, monthlyIncome=5000.0, monthlyDebt=1500.0, requestedAmount=200000.0, dti=30.0}
